yt_dlp/extractor/freetv.py

```diff
@@ -0,0 +1,176 @@
+
+        video_id = traverse_obj(api_response, ('data', 'displayMeta', 'contentID'), expected_type=str)
+        video_url = traverse_obj(api_response, ('data', 'displayMeta', 'streamURLVideo'), expected_type=str)
+
+        if determine_ext(video_url) != 'm3u8':
+            raise ExtractorError('Manifest was not found', video_id=video_id)
+
+        formats, subtitles = self._extract_m3u8_formats_and_subtitles(video_url, video_id, 'mp4')
+        self._sort_formats(formats)
+
+        return {
+            'id': video_id,
+            'title': traverse_obj(api_response, ('data', 'displayMeta', 'title')),
+            'description': traverse_obj(api_response, ('data', 'displayMeta', 'desc')),
+            'formats': formats,
+            'subtitles': subtitles,
+        }
+
+    def _extract_series(self, display_id, season_ids, series_title, series_description,
+                        action='olyott_get_dynamic_series_content'):
+        entries = []
+        for season_id in season_ids:
+            api_response = self._get_api_response(
+                season_id, resource_type='series',
+                postdata={
+                    'action': action,
+                    'contentID': season_id,
+                    'type': 'list',
+                    'perPage': '1000',
+                })
+
+            season_episodes = traverse_obj(api_response, ('data', '1'))
+
+            if season_episodes is None:
+                raise ExtractorError("Response doesn't contain episode list")
+
+            for episode in season_episodes:
+                video_id = str_or_none(episode.get('contentID'))
+                video_url = episode.get('streamURL')
+
+                if determine_ext(video_url) == 'm3u8':
+                    formats, subtitles = self._extract_m3u8_formats_and_subtitles(
+                        video_url, video_id, 'mp4', fatal=False)
+                    self._sort_formats(formats)
+
+                    entries.append({
+                        'id': video_id,
+                        'title': episode.get('fullTitle'),
+                        'description': episode.get('description'),
+                        'formats': formats,
+                        'subtitles': subtitles,
+                        'thumbnail': episode.get('thumbnail'),
+                        'series': series_title,
+                        'series_id': traverse_obj(episode, ('contentMeta', 'displayMeta', 'seriesID')),
+                        'season_id': traverse_obj(episode, ('contentMeta', 'displayMeta', 'seasonID')),
+                        'season_number': int_or_none(
+                            traverse_obj(episode, ('contentMeta', 'displayMeta', 'seasonNum'))),
+                        'episode_number': int_or_none(
+                            traverse_obj(episode, ('contentMeta', 'displayMeta', 'episodeNum'))),
+                    })
+                else:
+                    self.report_warning("Manifest was not found", video_id)
```

Comment:
create a generator and pass it as the entries to playlist_result

Reply:
I've created a function called `_extract_series_season`, does that work?